### PR TITLE
Publisher

### PR DESCRIPTION
**Pull request by glatard**
`bosh-publish` finds valid Boutiques descriptors in a local Git repository and generate OWL objects to be added to the [Neurolinks](https://brainhack101.github.io/neurolinks) index. It could be extended using [PyGihub](https://github.com/PyGithub/PyGithub) to automatically PR in Neurolinks.

### CODE REVIEW

File: tools/python/boutiques/neurolinks-template/tool.owl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+nl:neurodocker rdf:type owl:ObjectProperty ;
```
Comment (gkiar) [inline tools/python/boutiques/neurolinks-template/tool.owl]: `nl:neurodocker` was specific for a certain tool - we want to set a unique identifier of `nl:.....` for each new tool, and ensure that it is unique

File: tools/python/boutiques/neurolinks-template/tool.owl [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    rdfs:label "@@__LABEL__@@"^^xsd:string ;
+    obo:IAO_0000115 "@@__DESCRIPTION__@@"^^xsd:string ;
+    prov:wasCreatedBy "@@__AUTHOR__@@"^^xsd:string ;
+    dct:source "@@__URL__@@"^^xsd:string ;
+    nl:has_boutiques:true
```
Comment (gkiar) [inline tools/python/boutiques/neurolinks-template/tool.owl]: no semi-colon after this line, and I'm not sure about the `:` between `has_boutiques` and `true`, either... it didn't work at first attempt but I changed this and the above simultaneously so don't know what did the trick. I think this is a problem though.

**Comment by glatard** [comment]
sounds good! I'll wait for the JSON implementation.

**Comment by gkiar** [comment]
Updated `JSON` implementation for neurolinks, as described [here](https://github.com/brainhack101/neurolinks/blob/master/jsons/tool.json#L115-L124)

**Comment by gkiar** [comment]
Currently this is my situation when installing from your fork:
```
$ virtualenv myenv -p python3.6 & source myenv/bin/activate
$ pip install -e .
$ cd /gitrepo/with/boutiques/descriptors
$ bosh-publish
Traceback (most recent call last):
  File "/Users/greg/code/gkiar/boutiques/tools/python/env/boutiques3.6/bin/bosh-publish", line 11, in <module>
    load_entry_point('boutiques', 'console_scripts', 'bosh-publish')()
  File "/Users/greg/code/gkiar/boutiques/tools/python/boutiques/publisher.py", line 364, in main
    results.github_login, results.github_password, results.no_github).publish()
  File "/Users/greg/code/gkiar/boutiques/tools/python/boutiques/publisher.py", line 78, in __init__
    self.fix_github_credentials()
  File "/Users/greg/code/gkiar/boutiques/tools/python/boutiques/publisher.py", line 112, in fix_github_credentials
    username = self.get_from_stdin("GitHub login")
  File "/Users/greg/code/gkiar/boutiques/tools/python/boutiques/publisher.py", line 166, in get_from_stdin
    answer = raw_input(prompt)
NameError: name 'raw_input' is not defined
```

I get the same error when running from an unrelated dir and passing the location with the `-b` flag. For reference, I'm using this repo: https://github.com/neurodata/boutiques-tools/

**Comment by glatard** [comment]
@gkiar, thanks for the review, it should be fixed now!

**Pull request merged**